Update util.js

drygoatair opened the pull request:
When using [violentmonkey/generator-userscript](https://github.com/violentmonkey/generator-userscript) I noticed that not all lines from my `meta.js` were carried over to my userscript because:
1. I used tabs instead of spaces
2. I used a key without a value (`@noframes`)

To fix this, I made two small changes to the mapping function of  `const entries`:
1. changed `indexOf(' ')` to `search(/\s/)`, to support tabs
2. added a conditional for no-value keys `@unwrap` and `@noframes`

drygoatair committed:
Update util.js

Two small changes to 'entries':
- regexp search for whitespace, instead of search for ' ', to support tabs
- added a conditional for '@unwrap' and '@noframes' so that these will be seen as valid entries

gera2ld commented on src/util.js:
```diff
@@ -55,8 +55,14 @@ export function getMetadata(metaFileContent, additionalGrantList) {
       line = line.slice(3).trim();
-      const i = line.indexOf(' ');
-      if (i < 0) return;
+      const i = line.search(/\s/);
```
How about making it more general so we don't need to list keys without values?

```js
const matches = line.match(/^(\S+)(\s.*)?$/);
if (!matches) return;
const key = matches[1];
const value = (matches[2] || '').trim();
```

drygoatair replied:
That seems even better.

gera2ld reviewed (changes requested):
Thanks for the PR! I think we can make it simpler.

drygoatair committed:
Update util.js

Made suggested changes:
https://github.com/violentmonkey/rollup-plugin-userscript/pull/3#pullrequestreview-1713967433

drygoatair reviewed (commented):
I think this is it? I've never done a pull request before so I'm not sure if I'm doing this right 😅

gera2ld reviewed (approved):
LGTM, thank you so much!
